chains/evm/listener/handlers/deposit.go

```diff
@@ -0,0 +1,192 @@
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/sygmaprotocol/sygma-core/relayer/message"
+	"github.com/sygmaprotocol/sygma-inclusion-prover/chains/evm/abi"
+	"github.com/sygmaprotocol/sygma-inclusion-prover/chains/evm/listener/events"
+	evmMessage "github.com/sygmaprotocol/sygma-inclusion-prover/chains/evm/message"
+	"github.com/sygmaprotocol/sygma-inclusion-prover/chains/evm/util"
+)
+
+const (
+	EVMStateRootMessage message.MessageType = "EVMStateRootMessage"
+)
+
+type Client interface {
+	CallContext(ctx context.Context, target interface{}, rpcMethod string, args ...interface{}) error
+	FetchEventLogs(ctx context.Context, contractAddress common.Address, event string, startBlock *big.Int, endBlock *big.Int) ([]types.Log, error)
+	BlockByHash(ctx context.Context, hash common.Hash) (*types.Block, error)
+	TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
+}
+
+type StorageProof struct {
+	Proof []string `json:"proof"`
+}
+
+type AccountProof struct {
+	AccountProof []string       `json:"accountProof"`
+	StorageProof []StorageProof `json:"storageProof"`
+}
+
+type DepositEventHandler struct {
+	log              zerolog.Logger
+	client           Client
+	domainID         uint8
+	routerAddress    common.Address
+	routerABI        ethereumABI.ABI
+	slotIndex        uint8
+	genericResources []string
+	msgChan          chan []*message.Message
+}
+
+func NewDepositEventHandler(
+	domainID uint8,
+	client Client,
+	routerAddres common.Address,
+	slotIndex uint8,
+	genericResources []string,
+	msgChan chan []*message.Message) *DepositEventHandler {
+	routerABI, _ := ethereumABI.JSON(strings.NewReader(abi.RouterABI))
+	return &DepositEventHandler{
+		log:              log.With().Uint8("domainID", domainID).Logger(),
+		client:           client,
+		routerAddress:    routerAddres,
+		routerABI:        routerABI,
+		slotIndex:        slotIndex,
+		genericResources: genericResources,
+		msgChan:          msgChan,
+		domainID:         domainID,
+	}
+}
+
+func (h *DepositEventHandler) HandleDeposits(destination uint8, startBlock *big.Int, endBlock *big.Int, slot *big.Int) error {
+	deposits, err := h.fetchDeposits(destination, startBlock, endBlock)
+	if err != nil {
+		return err
+	}
+	msgs := make(map[uint8][]*message.Message)
+	for _, d := range deposits {
+		accountProof, storageProof, err := h.proof(endBlock, d)
+		if err != nil {
+			return err
+		}
+
+		h.log.Debug().Uint8("destination", d.DestinationDomainID).Msg("Sending transfer message")
+
+		msgs[d.DestinationDomainID] = append(msgs[d.DestinationDomainID], evmMessage.NewEVMTransferMessage(h.domainID, d.DestinationDomainID, evmMessage.TransferData{
+			Deposit:      d,
+			Slot:         slot,
+			AccountProof: accountProof,
+			StorageProof: storageProof,
+			Type:         h.transferType(d),
+		}))
+	}
+	if len(msgs) == 0 {
+		log.Warn().Msgf("No deposits found for block range %s-%s", startBlock, endBlock)
```

Comment:
Is this a warning log? I thought this would happen each time there weren't deposits in the block range.